sdv/data_processing/data_processor.py

```diff
@@ -69,3 +73,75 @@ def __init__(self, metadata, learn_rounding_scheme=True, enforce_min_max_values=
         self._constraints_to_reverse = []
         self._transformers_by_sdtype = self._DEFAULT_TRANSFORMERS_BY_SDTYPE.copy()
         self._update_numerical_transformer(learn_rounding_scheme, enforce_min_max_values)
+
+    def filter_valid(self, data):
+        """Filter the data using the constraints and return only the valid rows.
+
+        Args:
+            data (pandas.DataFrame):
+                Table data.
+
+        Returns:
+            pandas.DataFrame:
+                Table containing only the valid rows.
+        """
+        for constraint in self._constraints:
+            data = constraint.filter_valid(data)
+
+        return data
+
+    def to_dict(self):
+        """Get a dict representation of this DataProcessor.
+
+        Returns:
+            dict:
+                Dict representation of this DataProcessor.
+        """
+        return {
+            'metadata': copy.deepcopy(self.metadata.to_dict()),
+            'constraints_to_reverse': copy.deepcopy(self._constraints_to_reverse),
+            'model_kwargs': copy.deepcopy(self._model_kwargs)
+        }
+
+    @classmethod
+    def from_dict(cls, metadata_dict, learn_rounding_scheme=True, enforce_min_max_values=True):
```

Comment:
Since `to_dict` doesn't provide `_transformers_by_sdtype`, we have to pass `learn_rounding_scheme` and `enforce_min_max_values` to derive it from scratch.